lib/ansible/plugins/strategy/__init__.py

```diff
@@ -876,14 +876,30 @@ def _do_handler_run(self, handler, handler_name, iterator, play_context, notifie
         host_results = []
         for host in notified_hosts:
             if not iterator.is_failed(host) or play_context.force_handlers:
-                task_vars = self._variable_manager.get_vars(play=iterator._play, host=host, task=handler)
-                self.add_tqm_variables(task_vars, play=iterator._play)
-                self._queue_task(host, handler, task_vars, play_context)
+                if handler.action == 'meta':
+                    meta_name = handler.args.get('_raw_params', None)
+                    # FIXME filter others?
+                    if meta_name in ('flush_handlers'):
+                        display.error('%s not supported as a handler, skipping...' % meta_name)
+                        return_data  = {'msg': '%s not supported as a handler' % meta_name, 'skipped': True}
```

Comment:
instead of displaying an error and then skipping the task, just return a task failed error

Reply:
That seems to make more sense.